openedx/frontend-app-ecommerce
# feat: Add Receipt Page from ecommerce to MFE

[REV-2683](https://2u-internal.atlassian.net/browse/REV-2683).
We're moving the Receipt Page from ecommerce repo to this MFE for the [Theseus Project](https://2u-internal.atlassian.net/wiki/spaces/RS/pages/7963455/Theseus+Project).
Current receipt page template: https://github.com/edx/ecommerce/blob/947c9068928281d24a54a86ea6df3340309a4f57/ecommerce/templates/edx/checkout/receipt.html#L1
Related backend PRs: 
https://github.com/openedx/ecommerce/pull/3748
https://github.com/openedx/ecommerce/pull/3778
https://github.com/openedx/ecommerce/pull/3781

**Note:** for screenshots of all the different versions, pls refer to https://2u-internal.atlassian.net/l/cp/N8fuc5Py

**Receipt page before:**
![ReceiptPage_OLD_basic](https://user-images.githubusercontent.com/13632680/181372753-04c2573f-4ee2-472d-b54a-5948249ade19.png)


**Receipt page after:**
![ReceiptPage_basic](https://user-images.githubusercontent.com/13632680/181307890-da643d11-8bff-4e9b-9616-a1c9797fae44.png)


_Order History update tag-along:_
This is not part of this ticket, but Paragon's `Table` component will be deprecated, I'm updating the Order History page to use `DataTable` so that both Order History and Receipt Page have the same look for tables.
<img width="1278" alt="Screen Shot 2022-06-07 at 3 43 08 PM" src="https://user-images.githubusercontent.com/13632680/173113903-526c0686-4c70-4b79-ad90-260f793a1470.png">

<img width="1278" alt="Screen Shot 2022-06-07 at 3 42 35 PM" src="https://user-images.githubusercontent.com/13632680/173113917-dc22c580-cbca-4c83-bde4-394eb1f79e84.png">

**This PR:**
- Creates the receipt page 
- Updates Order History to use the new `DataTable` Paragon component
- Adds unit/snapshot tests

This PR is dependent on the backend work in https://2u-internal.atlassian.net/browse/REV-2788

**To test locally:**
1) Enroll in a course, upgrade and make a purchase
2) Go to http://localhost:1996/orders to make sure you have orders 
3) Choose an order number (ie. EDX-100005) and go to `http://localhost:1996/receipt/?order_number={order-number}`
4) To see a discount, a coupon must be created here http://localhost:18130/coupons/ and applied at checkout
5) This page CANNOT be accessed yet through the "View Order Details" link on Order History, this contains our current ecommerce receipt page
